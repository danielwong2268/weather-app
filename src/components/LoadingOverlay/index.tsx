import React from 'react';
import styled from 'react-emotion';
import Spinner from '../Spinner';
import { WHITE } from 'src/constants/colors';

const Overlay = styled('div')`
  opacity: 0.55;
  background-color: ${WHITE};
  position: absolute;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 10;
  top: 0;
  left: 0;
`;

const LoadingOverlay = () => (
  <Overlay>
      <Spinner />
  </Overlay>
);

export default LoadingOverlay;