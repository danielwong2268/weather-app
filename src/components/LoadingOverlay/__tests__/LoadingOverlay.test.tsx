import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LoadingOverlay from '..';

describe('<FadeTransition />', () => {
  it('Renders an the overlay', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<LoadingOverlay />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});