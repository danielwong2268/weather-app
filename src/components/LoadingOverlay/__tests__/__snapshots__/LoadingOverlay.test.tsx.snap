// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FadeTransition /> Renders an the overlay 1`] = `
<Styled(div)>
  <Spinner />
</Styled(div)>
`;
